aya/src/programs/tc.rs

```diff
@@ -120,12 +124,13 @@ impl SchedClassifier {
         &mut self,
         interface: &str,
         attach_type: TcAttachType,
+        priority: u32,
```

Comment:
I think this should be `u16` in the API if the valid range is `0..u16::MAX`, then we should cast internally.

Reply:
@alessandrod thanks for the review.  I went back and forth on this myself.  Using a u16 requires more code changes and conversions back and forth, so I ended up keeping it as a u32 in the API.  However, making it a u16 is probably best for the user, so I'll make that change.

Reply:
The question I had was whether to just change priority to u16 in the SchedClassifier attach API, or everywhere in tc.rs and netlink.rs.  I chose everywhere to keep it consistent.  What do you think?

Reply:
FWIW, the iproute2 code has priority/preference as a u32 and does a range check:
https://github.com/shemminger/iproute2/blob/main/tc/tc_filter.c#L150

Reply:
Do you know if there's a rationale for this? Is the range expected to be increased?

Reply:
Not that I know of.  They also do a <<16 [here](https://github.com/shemminger/iproute2/blob/main/tc/tc_filter.c#L186) like we're doing, so only the lower 16 bits are used anyway.  I think it's more user-friendly for us to make it a u16 at least in the API.

Reply:
Yep agreed!